### PR TITLE
[2.7] bpo-29165: doc: make extending/newtypes more Python 3 friendly

### PR DESCRIPTION
**Pull request by methane**

### CODE REVIEW

File: Doc/extending/newtypes.rst [hunk 1]
```diff
@@ -111,23 +110,16 @@ it's common practice to not specify them explicitly unless you need them.
 This is so important that we're going to pick the top of it apart still
 further::
 
-   PyObject_HEAD_INIT(NULL)
+   PyVarObject_HEAD_INIT(NULL, 0)
 
 This line is a bit of a wart; what we'd like to write is::
 
-   PyObject_HEAD_INIT(&PyType_Type)
+   PyVarObject_HEAD_INIT(&PyType_Type, 0)
```
Comment (serhiy-storchaka) [inline Doc/extending/newtypes.rst:117]: It is worth to explain what `0` means here (it is a historical artifact).

File: Doc/extending/newtypes.rst [hunk 2]
```diff
@@ -244,7 +236,7 @@ doesn't do anything. It can't even be subclassed.
 Adding data and methods to the Basic example
 --------------------------------------------
 
-Let's expend the basic example to add some data and methods.  Let's also make
+Let's extend the basic example to add some data and methods.  Let's also make
```
Comment (serhiy-storchaka) [inline Doc/extending/newtypes.rst:239]: "extend" or "expand"?

File: Doc/extending/newtypes.rst [hunk 3]
```diff
@@ -965,14 +957,15 @@ Finalization and De-allocation
 
 This function is called when the reference count of the instance of your type is
 reduced to zero and the Python interpreter wants to reclaim it.  If your type
-has memory to free or other clean-up to perform, put it here.  The object itself
-needs to be freed here as well.  Here is an example of this function::
+has memory to free or other clean-up to perform, you can put it here.  The
```
Comment (serhiy-storchaka) [inline Doc/extending/newtypes.rst:960]: What is changed in these lines?

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]

File: Doc/extending/newtypes.rst (hunk 1 above)
Comment (methane) [reply Doc/extending/newtypes.rst:117]: I want this pull request only copies from Python 3's doc.

File: Doc/extending/newtypes.rst (hunk 2 above)
Comment (methane) [reply Doc/extending/newtypes.rst:239]: "extend"
https://docs.python.org/3/extending/newtypes.html#adding-data-and-methods-to-the-basic-example

File: Doc/extending/newtypes.rst (hunk 3 above)
Comment (methane) [reply Doc/extending/newtypes.rst:960]: Added "you can".
https://docs.python.org/3/extending/newtypes.html#finalization-and-de-allocation

**Comment by methane** [comment]
I checked diff of Doc/include/*.c between 2.7 and master, and backported many style fixes.

I checked ``$ gcc `python2-config --cflags` shoddy.c -c`` for noddy.c, noddy2.c, ... shoddy.c,
and there are no warnings.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by methane** [comment]
thanks

**Pull request merged**